hs/src/Reach/Simulator/Core.hs

```diff
@@ -521,9 +558,21 @@ instance Interp LLStep where
       _ <- interp stmt
       interp step
     LLS_Stop _at -> return V_Null
-    LLS_ToConsensus _at _lct _tc_send tc_recv _tc_mtime -> do
-      -- TODO: block thread for race winner
-      interp $ dr_k tc_recv
+    LLS_ToConsensus _at _lct tc_send (DLRecv {..}) _tc_mtime -> do
+      v <- suspend $ PS_Suspend (A_TieBreak $ M.keys $ M.mapKeys bunpack tc_send)
+      case v of
+        V_UInt n -> do
+          g <- globalGet
+          let locals = e_locals g
+          let lclst = l_who (locals M.! (fromIntegral n))
+          case lclst of
+            Nothing ->  impossible "expected participant id, received consensus id"
+            Just part -> case tc_send M.! part of
+              DLSend {..} -> do
+                ds_msg' <- mapM interp ds_msg
```

Comment:
The way you have this won't work in the long-run.

First, simple thing, L572 is interpreting the messages with the consensus store, but they will only be bound in the participant store.

Second, L562 is assuming that anyone can be chosen, but actually there might be a situation where Alice and Bob are "racing", but actually Bob hasn't started the program at all, so you're going to have to track in some way who is actually available.

Third, L567 means that the identifier is the participant id, not the the index into the list in L562. You probably know this, but I would be confused and forget.

Fourth, you're going to need to have some sort of GlobalState counter of which message this is and save it for later in L572. Because, you're going to have to go back to e.g. Alice and run the same `dr_k` in her context, but then maybe 10 years later, when someone runs another instance of Bob, he'll need to see that same publish and run the code too. There's got to be some sort of global log of every call to the contract.

Reply:
"Second, L562 is assuming that anyone can be chosen, but actually there might be a situation where Alice and Bob are "racing", but actually Bob hasn't started the program at all, so you're going to have to track in some way who is actually available."